### ISSUE TITLE
Decide on three algorithms

### ISSUE THREAD

**Issue by psvc** [psvc/fraud_detection_fatema_piotr#18]

**Comment by psvc** [comment]
Light Gradient Boosting (LGB), Random Forest (RF), and eXtreme Gradient Boosting (XGB) 
<google-sheets-html-origin><style type="text/css"></style>
Feature | Light Gradient Boosting (LGBM) | Random Forest (RF) | eXtreme Gradient Boosting (XGBoost)
-- | -- | -- | --
Algorithm Type | Gradient Boosting (leaf-wise) | Ensemble (Bagging) of Decision Trees | Gradient Boosting (level-wise)
Speed | Very Fast | Moderate to Slow (depending on dataset size) | Fast (but generally slower than LGBM on large datasets)
Memory Usage | Low | Moderate to High | Moderate to High (generally higher than LGBM)
Overfitting Risk | Higher (especially on small datasets, if not tuned) | Low | Low (due to regularization)
Performance | Often state-of-the-art, especially on large datasets | Very good, robust | Often state-of-the-art
Imbalanced Data | Excellent (GOSS, can use is_unbalance=True) | Good (requires resampling techniques like SMOTE) | Excellent (scale_pos_weight, can use resampling techniques)
Categorical Features | Native handling | Requires encoding (e.g., one-hot encoding) | Requires encoding (e.g., one-hot encoding), unless specific implementation
Interpretability | Feature importance, but individual tree decisions are complex | Feature importance, generally good for understanding | Feature importance, but individual tree decisions are complex
Scalability | Highly scalable, supports GPU | Scalable, can be parallelized | Highly scalable, supports GPU

**Comment by psvc** [comment]
@Faes404 as per our discussion, created subtasks with each alghoritm being a task, add another one if none of those is what you want to do. We just need a skeleton = of how we are going to use this to perform better than a threshold and make it 'working'. We can fix the data distribution later.

**Comment by psvc** [comment]
Assign it to you just for visibility when you take one of the subtasks, assign to yourself.
